src/Voron/Impl/Transaction.cs

```diff
@@ -86,6 +86,7 @@ public void Commit()
 
             PrepareForCommit();
             _lowLevelTransaction.Commit();
+            _lowLevelTransaction.Environment.CurrentWriteTransaction = null;
```

Comment:
This assumes that no one ever uses the LowLevelTransaction directly, which is wrong.